signing_service/signing_service/tests/test_authenticate.py

```diff
@@ -0,0 +1,157 @@
+from contextlib import closing
+import os
+import socket
+
+from golem_messages.cryptography import ECCx
+from golem_messages.cryptography import ecdsa_verify
+import assertpy
+import mock
+import pytest
+
+from middleman_protocol.message import AbstractFrame
+from middleman_protocol.message import AuthenticationChallengeFrame
+from middleman_protocol.message import AuthenticationResponseFrame
+from middleman_protocol.stream import unescape_stream
+
+from signing_service.constants import SIGNING_SERVICE_RECOVERABLE_ERRORS
+from signing_service.signing_service import SigningService
+from .utils import SigningServiceIntegrationTestCase
+
+
+TEST_ETHEREUM_PRIVATE_KEY = '3a1076bf45ab87712ad64ccb3b10217737f7faacbf2872e88fdd9a537d8fe266'
+
+concent_ecc_keys = ECCx(None)
+(CONCENT_PRIVATE_KEY, CONCENT_PUBLIC_KEY) = concent_ecc_keys.raw_privkey, concent_ecc_keys.raw_pubkey
+
+signing_service_ecc_keys = ECCx(None)
+(SIGNING_SERVICE_PRIVATE_KEY, SIGNING_SERVICE_PUBLIC_KEY) = signing_service_ecc_keys.raw_privkey, signing_service_ecc_keys.raw_pubkey
+
+
+class TestSigningServiceAuthenticate(SigningServiceIntegrationTestCase):
+
+    host = None
```

Comment:
That would be awesome because mypy is complaining about this construction (it gets confused that is is declared as `None` first then gets different type).

Reply:
:+1: done